src/System.IO.Pipelines/StreamExtensions.cs

```diff
@@ -78,9 +64,18 @@ public static IPipeReader AsPipelineReader(this Stream stream, CancellationToken
         /// <param name="stream"></param>
         /// <param name="writer"></param>
         /// <returns></returns>
-        public static Task CopyToAsync(this Stream stream, IPipeWriter writer)
+        public static async Task CopyToAsync(this Stream stream, IPipeWriter writer)
         {
-            return stream.CopyToAsync(new PipelineWriterStream(writer));
+            try
+            {
+                await stream.CopyToAsync(new PipelineWriterStream(writer));
+            }
+            catch (Exception ex)
+            {
+                writer.Complete(ex);
+                return;
+            }
+            writer.Complete();
```

Comment:
Interesting point. I think we should definitely complete the writer in the case of an error, but I can see the argument for leaving the pipe open after a successful copy by default. What I don't like is calling Complete() both here and ExecuteCopyToAsync(). Pick one.